components/concept/push/src/main/java/mozilla/components/concept/push/PushProcessor.kt

```diff
@@ -6,74 +6,95 @@
 
 package mozilla.components.concept.push
 
-import androidx.annotation.CallSuper
+import androidx.annotation.VisibleForTesting
 
 /**
  * A push notification processor that handles registration and new messages from the [PushService] provided.
  * Starting Push in the Application's onCreate is recommended.
  */
-abstract class PushProcessor {
+interface PushProcessor {
 
     /**
-     * Initialize the PushProcessor before it
+     * Start the push processor and any service associated.
      */
-    @CallSuper
-    open fun initialize() {
-        instance = this
-    }
-
-    /**
-     * Start the push processor and starts any service associated.
-     */
-    abstract fun start()
+    fun initialize()
 
     /**
-     * Stop the push service and stops any service associated.
+     * Removes all push subscriptions from the device.
      */
-    abstract fun stop()
+    fun shutdown()
```

Comment:
rename this (`unsubscribeAll`?), current name implies that this is opposite to `initialize` and is part of the app lifecycle.

Reply:
Also, I think you want this to return `Deferred<Unit>`, so that consumers can block (if they want to) on the results of this. E.g. if we want to ensure that we've done with this operation before doing other cleanup.

Reply:
> rename this (unsubscribeAll?), current name implies that this is opposite to initialize and is part of the app lifecycle.

Since we're stopping the push service as well, we figured that this should be called something more dramatic. I've updated the KDocs, but did we still want to change it to "unsubscribeAll" since it does more than that.

> Also, I think you want this to return Deferred<Unit>, so that consumers can block (if they want to) on the results of this. E.g. if we want to ensure that we've done with this operation before doing other cleanup.

I'll look this up to verify, but I don't think I can since I'm cancelling the job right after.